### DISCUSSION TITLE
DataGrid filter not working on data load?

### DISCUSSION THREAD

**Discussion by MichaelMcGlothlin** [Megabit/Blazorise#4134] [Q&A]
I have a SelectList that I’m using to filter a DataGrid. That part works fine. But I’ve been asked to make it initially filter by a value that isn’t the SelectList default “All”. I am setting that SelectList value fine but it doesn’t seem to do anything. I don’t have a context
to call TriggerFilterChange() and StateHasChanged() doesn’t seem to do anything. Any idea what I might be missing?

**Comment by David-Moreira** [comment]
Hello.
Can you try looking at https://github.com/Megabit/Blazorise/blob/master/Demos/Blazorise.Demo/Pages/Tests/DataGrid/DataGridPage.razor
There should be a working example here. If you're still not able to get it to work, can you post a sample of the 'faulty' code? Thanks!
